test/plugin/base_test.rb

```diff
@@ -1160,11 +1342,27 @@ def setup_prometheus
     Prometheus::Client.registry.instance_variable_set('@metrics', {})
   end
 
+  def url_with_local_resource_id(local_resource_id)
+    "#{DEFAULT_METADATA_AGENT_URL}/monitoredResource/#{local_resource_id}"
+  end
+
+  def setup_metadata_agent_stubs
+    MONITORED_RESOURCE_STUBS.each do |local_resource_id, resource|
+      stub_request(:get, url_with_local_resource_id(local_resource_id))
+        .to_return(status: 200, body: resource)
+    end
+  end
+
+  def assert_requested_metadata_agent_stub(local_resource_id)
+    assert_requested :get, url_with_local_resource_id(local_resource_id)
+  end
+
   def container_tag_with_container_name(container_name)
     "kubernetes.#{CONTAINER_POD_NAME}_#{CONTAINER_NAMESPACE_NAME}_" \
       "#{container_name}"
   end
 
+  # GKE Container
```

Comment:
(a) Should this be above `container_tag_with_container_name`?
(b) Missing trailing '.'.

Reply:
Moved this above and added `.`.